hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextQueryProcessor.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec;
+
+import static com.hedera.node.app.service.contract.impl.hevm.HederaEvmVersion.EVM_VERSIONS;
+
+import com.hedera.node.app.service.contract.impl.annotations.QueryScope;
+import com.hedera.node.app.service.contract.impl.hevm.ActionSidecarContentTracer;
+import com.hedera.node.app.service.contract.impl.hevm.HederaEvmContext;
+import com.hedera.node.app.service.contract.impl.hevm.HederaEvmTransaction;
+import com.hedera.node.app.service.contract.impl.hevm.HederaEvmVersion;
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater;
+import com.hedera.node.app.service.contract.impl.infra.HevmStaticTransactionFactory;
+import com.hedera.node.app.service.contract.impl.state.ProxyWorldUpdater;
+import com.hedera.node.app.spi.workflows.QueryContext;
+import com.hedera.node.config.data.ContractsConfig;
+import com.swirlds.config.api.Configuration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+import javax.inject.Inject;
+
+/**
+ * A utility class for running
+ * {@link TransactionProcessor#processTransaction(HederaEvmTransaction, HederaWorldUpdater, Supplier, HederaEvmContext, ActionSidecarContentTracer, Configuration)}
+ * call implied by the in-scope {@link QueryContext}.
+ * Analogous to {@link ContextTransactionProcessor} but for queries.
+ */
+@QueryScope
+public class ContextQueryProcessor implements Callable<CallOutcome> {
+    private final QueryContext context;
+    private final Configuration configuration;
+    private final ContractsConfig contractsConfig;
+    private final HederaEvmContext hederaEvmContext;
+    private final ActionSidecarContentTracer tracer;
+    private final ProxyWorldUpdater worldUpdater;
+    private final HevmStaticTransactionFactory hevmStaticTransactionFactory;
+    private final Supplier<HederaWorldUpdater> feesOnlyUpdater;
+    private final Map<HederaEvmVersion, TransactionProcessor> processors;
+
+    @Inject
+    public ContextQueryProcessor(
+            @NonNull final QueryContext context,
+            @NonNull final Configuration configuration,
+            @NonNull final ContractsConfig contractsConfig,
```

Comment:
Again, no need to use dagger here. Instead of passing in the configuration, use context.configuration() to get the configurations needed inside of call() instead.